api/src/utils/xlsx-utils/worksheet-utils.ts

```diff
@@ -321,3 +321,27 @@ export function validateCsvFile(xlsxWorksheet: xlsx.WorkSheet, columnValidator:
 
   return true;
 }
+
+/**
+ * This function pulls out any non-standard columns from a CSV so they can be processed separately.
+ *
+ * @param {xlsx.WorkSheet} xlsxWorksheet The worksheet to pull the columns from
+ * @returns {*} string[] The list of non-standard columns found in the CSV
+ */
+export function getNonStandardColumnNamesFromWorksheet(
+  xlsxWorksheet: xlsx.WorkSheet,
+  columnValidator: IXLSXCSVValidator
+): string[] {
+  const columns = getHeadersUpperCase(xlsxWorksheet);
+
+  let aliasColumns: string[] = [];
+  // Create a list of all column names and aliases
+  if (columnValidator.columnAliases) {
+    aliasColumns = Object.values(columnValidator.columnAliases).flat();
+  }
+
+  const standardColumNames = [...columnValidator.columnNames, ...aliasColumns];
```

Comment:
Nice!

Reply:
I wonder if we should add a comment to the `IXLSXCSVValidator` interface, to specify that columnNames and columnAliases should be uppercase?